pkg/kubectl/resource/mapper.go

```diff
@@ -54,19 +56,21 @@ func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to decode %q: %v", source, err)
 	}
-	var obj runtime.Object
-	var versioned runtime.Object
-	if isThirdParty, gvkOut, err := thirdpartyresourcedata.IsThirdPartyObject(data, gvk); err != nil {
-		return nil, err
-	} else if isThirdParty {
-		obj, err = runtime.Decode(thirdpartyresourcedata.NewDecoder(nil, gvkOut.Kind), data)
-		versioned = obj
-		gvk = gvkOut
-	} else {
-		obj, versioned = versions.Last(), versions.First()
-	}
-	if err != nil {
-		return nil, fmt.Errorf("unable to decode %q: %v [%v]", source, err, gvk)
+
+	obj, versioned := versions.Last(), versions.First()
+	if !m.dynamic {
```

Comment:
The codecs provided by the factory come from `ClientForMapping` whose only input is the GVK. There is code in there to handle third party based on the GVK and use a third party codec instead.

My needs are slightly different. If the caller wants to use dynamic decoding, that is the only codec they should use, regardless of the GVK. But I also need to leave the orginal codecs intact for commands that can't use the dynamic objects yet. (like describe, delete, ....). So the choice to use dynamic objects here isn't based on the mapping, but rather if the caller doesn't need the versioned types.